### ISSUE TITLE
Native Firefox’s interface capture the input in opening

### ISSUE THREAD

**Issue by bepo-keyboard** [tridactyl/tridactyl#4105]
### Brief description of the problem:
When I open Firefox, the native interface of Firefox (ie. the native navbar) immediately get the input focus of keyboard, and I need to click with the cursor inside the page area of the Firefox interface until I get the Tridactyl’s behavior. The same annoying behavior occur when I use <kbd>CTRL</kbd>-<kbd>T</kbd> instead of Tridactyl’s `:tabnew` command.

And this also occure when I hide the `#nav-bar` with the following line in the `userChrome.css` :

```
#nav-bar{ height: 0px !important; min-height: 0px !important; visibility: hidden;}
```

The navbar still invisible, but the Tridactyl’s interface didn’t get the inputs. Like if the cursor was in the invisible navbar.

#### History
Before this hacks, I tried to hide the navbar with `:guiset gui none`. But it doesn’t work. It have a strange behavior similar to the one I describe now.

### Steps to reproduce:
1. Open Firefox ;
2. Try to use any Tridactyl’s command (such `:someting`).

### Tridactyl version (`:version`):

1.22.0

### Firefox version :

Firefox 91.6.0esr over Debian 11

###  `:viewconfig --user`
https://gist.github.com/bepo-keyboard/e6b2403f5d6edfe5ea6b0ef011250c57

###  `~/.tridactylrc` 

Noting interesting exept some rebindings for BÉPO keyboard.
```
" vim: set filetype=tridactyl

" Retour à la configuration par défaut
sanitise tridactyllocal tridactylsync

"
keymap t j
keymap s k
keymap c h
keymap r l

keymap C H
keymap R L


keymap j t
keymap J T

keymap k s
keymap K S

bind i tabprev
bind e tabnext

set editorcmd urxvt -e "vim %f '+normal!%lGzv%c|'"
```

**Comment by bovine3dom** [comment]
Hi, unfortunately this is a Firefox decision and not something that we can fix.

You can:

- make your own bind (but not to Ctrl-T which is reserved by Firefox) with e.g. `:bind --mode=browser <CS-t> tabopen` which would leave you with focus on the web page
- just use `t` instead of making a new tab and then choosing where to go with it
- press Ctrl+comma to get focus back to Tridactyl from almost anywhere in the browser
- or you can patch Firefox to stop reserving keys here https://github.com/alerque/que-firefox/blob/master/bin/patch-firefox.sh and then bind tabopen to control-t

Hope this helps :)

**Comment by bepo-keyboard** [comment]
<emotiv-talk>
Thank you for your reactivity, it’s great the way you’re managing Tridactyl project.
Tridactyl is a great add-on and it restore my hope since Vimperator died :)
</emotiv-talk>

> Hi, unfortunately this is a Firefox decision and not something that we can fix.

Their is also no way to fix it by acting directly on Firefox itself, in `about:config` or something else?

> You can:
> 
>     * make your own bind (but not to Ctrl-T which is reserved by Firefox) with e.g. `:bind --mode=browser <CS-t> tabopen` which would leave you with focus on the web page
> 
>     * just use `t` instead of making a new tab and then choosing where to go with it
> 
>     * press Ctrl+comma to get focus back to Tridactyl from almost anywhere in the browser
> 
>     * or you can patch Firefox to stop reserving keys here https://github.com/alerque/que-firefox/blob/master/bin/patch-firefox.sh and then bind tabopen to control-t
All this stuffs are usefull. Anyway, I can forget <kbd>ctrl</kbd>-<kbd>T</kbd>, i’m not really bounded to this binding. But my main question is how to get the focus on the main page at the Fiferox start.

Since I have the focus from the Firefox opening, I can avoid  <kbd>ctrl</kbd>-<kbd>T</kbd>.

**Comment by bovine3dom** [comment]
Hmm, I get focus on the page when I start Firefox but I don't remember changing any settings. Maybe it's because I am starting Firefox with some tabs open and it isn't a new tab page?

**Comment by bepo-keyboard** [comment]
> Hmm, I get focus on the page when I start Firefox but I don't remember changing any settings. **Maybe it's because I am starting Firefox with some tabs open and it isn't a new tab page?**

Yes you’re right. When I start firefox with an alredy opened tabs I don’t need to do <kbd>ctrl</kbd>-<kbd>,</kbd>.

**Comment by bovine3dom** [comment]
Maybe try starting Firefox with e.g. `firefox wikipedia.org` instead so that a URL is specified?

**Comment by Mckol** [comment]
Hi, [this extension](https://addons.mozilla.org/en-US/firefox/addon/new-tab-override/) seems to be able to make the browser focus the page instead so it should be possible

**Comment by bovine3dom** [comment]
Oh, thanks, you're right - they have a hack where they open a new tab and close the old tab here:

https://github.com/cadeyrn/newtaboverride/blob/a45c91faa420dbcbeb14e93ecdacf0fd031e6185/src/js/core/newtab.js#L77-L100

**Comment by Nimmidev** [comment]
@bepo-keyboard i may have a suggestion how you can use what bovine3dom has suggested (`:bind --mode=browser <CS-t> tabopen`) while still using the `<C-t>` keyboard shortcut.

This only works if you have enabled firefox [autoconfig](https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig#w_setting-up-autoconfig) feature. In addition to that you need to disable the sandboxing of the autoconfig feature by 
 adding `pref("general.config.sandbox_enabled", false);` to the `autoconfig.js` file. More infos about that [here](https://www.userchrome.org/what-is-userchrome-js.html).

If you successfuly enabled that feature. You can add the following code to the `firefox.cfg` file, after the firefox window has loaded to disable the `<C-t` shortcut:

```
const keybind = window.document.querySelector("key_newNavigatorTab");
keybind.parentElement.removeChild(keybind);
```

Thats how i did it and it works great.
